File: src/Unidesk/Unidesk/Dtos/StudyProgrammeDto.cs
using System.ComponentModel.DataAnnotations;
using FluentValidation;
using Unidesk.Validations;

namespace Unidesk.Dtos;

public class StudyProgrammeDto : TrackedEntityDto, IValidatedEntity<StudyProgrammeDto>

{
    [Required]
    public string NameEng { get; set; }
    
    [Required]
    public string NameCze { get; set; }
    
    [Required]
    public string? Code { get; set; }
    
    [Required]
    public string? DescriptionEng { get; set; }
    
    [Required]
    public string? DescriptionCze { get; set; }

    public void ValidateAndThrow(StudyProgrammeDto item) => new StudyProgrammeDtoValidation().ValidateAndThrow(item);
}

File: src/Unidesk/Unidesk.Client/src/api-client/models/StudyProgrammeDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { TrackedEntityDto } from './TrackedEntityDto';

export type StudyProgrammeDto = (TrackedEntityDto & {
nameEng: string;
nameCze: string;
code: string;
descriptionEng: string;
descriptionCze: string;
});
